llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -13714,6 +13739,28 @@ void RISCVTargetLowering::ReplaceNodeResults(SDNode *N,
     // sext_inreg we emit for ADD/SUB/MUL/SLLI.
     LoadSDNode *Ld = cast<LoadSDNode>(N);
 
+    if (N->getValueType(0) == MVT::i64) {
+      assert(Subtarget.hasStdExtZilsd() && !Subtarget.is64Bit() &&
+             "Unexpected custom legalisation");
+
```

Comment:
Do you need to check if this is not an extending load, like you checked the store wasn't a truncating store?

Reply:
Line 13735

Reply:
D'oh, thanks